services/skus/service.go

```diff
@@ -1228,10 +1228,28 @@ func (s *Service) GetSingleUseCreds(ctx context.Context, orderID, itemID, reqID
 //
 // Browser's api_request_helper does not understand Go's nil slices, hence explicit empty slice is returned.
 func (s *Service) GetTimeLimitedV2Creds(ctx context.Context, orderID, itemID, reqID uuid.UUID) ([]TimeAwareSubIssuedCreds, int, error) {
+	creds, err := s.Datastore.GetTLV2Creds(ctx, s.Datastore.RawDB(), orderID, itemID, reqID)
+	if err != nil {
+		return []TimeAwareSubIssuedCreds{}, http.StatusInternalServerError, fmt.Errorf("error getting credentials: %w", err)
+	}
+
+	activeCreds := filterActiveCreds(creds.Credentials, time.Now().UTC())
+	if activeCreds != nil {
+		return activeCreds, http.StatusOK, nil
+	}
+
 	obmsg, err := s.Datastore.GetSigningOrderRequestOutboxByRequestID(ctx, s.Datastore.RawDB(), reqID)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
-			return []TimeAwareSubIssuedCreds{}, http.StatusNotFound, errLegacyOutboxNotFound
+			// We have no credentials (active or expired) and an empty outbox return a 404.
+			if len(creds.Credentials) == 0 && activeCreds == nil {
+				return []TimeAwareSubIssuedCreds{}, http.StatusNotFound, errLegacyOutboxNotFound
+			}
+
+			// Nothing is being signed and all the creds are expired.
+			// This is to address and issue with switching to MDR where we deleted all the creds
+			// from the outbox and not just legacy.
+			return []TimeAwareSubIssuedCreds{}, http.StatusOK, nil
```

Comment:
do we need these changes anymore? I think the 200 case is already covered above with the db retrieval / filtering changes